### PR TITLE
Initial import of TAP controllers

### PR DESCRIPTION
**Pull request by LorenzMeier**
This gives us serial motor control.

### CODE REVIEW

**Comment by mhkabir** [comment]
What ESCs are these? Are they available to buy?

File: src/drivers/tap_esc/drv_tap_esc.h [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	uint8_t data[100];
+	uint8_t crc_data;
+
+} ESC_FEEDBACK_PACKET;
+
+typedef  struct {
+	uint8_t head;
+	uint8_t tail;
+	uint8_t dat_cnt;
+	uint8_t esc_feedbacck_buf[128];
```
Comment (dagar) [inline src/drivers/tap_esc/drv_tap_esc.h]: typo esc_feedbacck_buf

**Comment by LorenzMeier** [comment]
@mhkabir We're hoping to have this eventually available from an OEM as alternative to the Intrinsyc provided ESCs. Right now they're not available yet.

**Comment by LorenzMeier** [comment]
Jenkins test this please.

**Comment by LorenzMeier** [comment]
Rebased and applied.

**Pull request closed without merging**